src/include/codegen/aggregation.h

```diff
@@ -37,20 +39,39 @@ namespace codegen {
 class Aggregation {
  public:
   // Setup the aggregation to handle the provided aggregates
-  void Setup(CodeGen &codegen,
+  void Setup(CompilationContext &context,
+             const std::vector<planner::AggregatePlan::AggTerm> &agg_terms,
+             bool is_global, std::vector<type::Type> &grouping_ai_types);
+
+  // Setup the aggregation to handle the provided aggregates
+  void Setup(CompilationContext &context,
```

Comment:
During `Setup`, pull out `RuntimeState` and assign it as a member variable. That way, you don't need to pass around the `CompilationContext` to each method.

Reply:
Many methods start with 
```c++
auto &codegen = context.GetCodeGen();
auto &runtime_state = context.GetRuntimeState();
```

Should I assign both references as member variable or assign codegen and call the two getters in each method?

Reply:
Just assign `runtime_state`, and pass around the `codegen` instance through methods. We pass around `codegen` to indicate that a given method is generating code. We pass around the `CompilationContext` during class instantiation for classes to pull out the pieces they need.